crates/router/src/core/admin.rs

```diff
@@ -1261,3 +1274,28 @@ pub(crate) fn validate_auth_type(
         }
     }
 }
+
+#[cfg(feature = "dummy_connector")]
+async fn validate_dummy_connector_enabled(
+    state: &AppState,
+    connector_name: &api_enums::Connector,
+) -> Result<(), errors::ApiErrorResponse> {
+    #[cfg(feature = "dummy_connector")]
+    if !state.conf.dummy_connector.is_enabled {
+        match connector_name {
+            api_enums::Connector::DummyConnector1
+            | api_enums::Connector::DummyConnector2
+            | api_enums::Connector::DummyConnector3
+            | api_enums::Connector::DummyConnector4
+            | api_enums::Connector::DummyConnector5
+            | api_enums::Connector::DummyConnector6
+            | api_enums::Connector::DummyConnector7 => {
+                return Err(errors::ApiErrorResponse::InvalidRequestData {
+                    message: format!("{} is not enabled", connector_name),
```

Comment:
Can we change the message here. This message provides significant information about the feature being present in one case and in other. Instead when disabled it shouldn't reveal the existence of this behaviour.

Something like
`invalid connector name`?